### PR TITLE
Added constructor that takes raw context

### PR DESCRIPTION
**Pull request by mehdi-taabouri**

### CODE REVIEW

File: include/zq/context.hpp [hunk 1]
```diff
@@ -172,6 +175,21 @@ namespace zq {
...
+   */
+  [[nodiscard]] auto inline mk_context(
+      void *ctx) noexcept -> tl::expected<Context, std::runtime_error>
+  {
+    if (ctx == nullptr) {
+      return tl::make_unexpected(std::runtime_error("Invalid context"));
+    }
+    return Context{ctx};
+  }
+
```
Comment (hmf-elekta) [inline include/zq/context.hpp]:
```suggestion
  /**
   * @brief Factory function for creating a context
   *  This version takes a raw pointer to a native zmq context
   * @param z_ctx Native zmq context
   * @return A \ref Context on success, error on failure.
   */
  [[nodiscard]] auto inline mk_context(
      void *z_ctx) noexcept -> tl::expected<Context, std::runtime_error>
  {
    if (z_ctx == nullptr) {
      return tl::make_unexpected(std::runtime_error("Invalid context"));
    }
    return Context{z_ctx};
  }

```

**Review by hmf-elekta — APPROVED** [review]

**Comment by hmf-elekta** [comment]
We should add a test too.

File: include/zq/context.hpp [hunk 2]
```diff
@@ -34,19 +34,25 @@ namespace zq {
...
     friend tl::expected<Context, ErrMsg> mk_context(
         ContextOptions auto const& options) noexcept;
 
+    friend tl::expected<Context, std::runtime_error> mk_context(
+        void* ctx) noexcept;
+
     void* z_ctx{nullptr};
 
-    Context(void* ctx) noexcept : z_ctx{ctx} {};
+    Context(void* ctx, bool owns_ctx = true) noexcept : z_ctx{ctx}, m_owns_ctx{owns_ctx} {};
```
Comment (hmf-elekta) [inline include/zq/context.hpp:46]: Do we need the `owns_ctx` as a constructor parameter? I think the constructor that takes a native context could set `m_owns_ctx` to `false` always.

Comment (mehdi-taabouri) [reply include/zq/context.hpp:46]: There is only one constructor which takes a pointer to a native context, which is used by the original `mk_context` functions.

**Pull request merged**